4-1.cpp

```diff
@@ -0,0 +1,151 @@
+#include <iostream>
+#include <cstdlib>
+
+using namespace std;
+
+// Вывод массива
+// iSize - размер массива
+// piArray - указатель на массив
+void vGetArray(int iSize, int* piArray) {
+	cout << "Итоговый массив: ";
+	for (int i = 0; i < iSize; i++) {
+		cout << piArray[i] << " ";
+	}
+	cout << endl;
+}
+
+// Заполнение массива
+void vSetArray(int iSize, int* piArray) {
+
+	int iChoice = 0;
+	cout << "Заполнение массива" << endl
+		 << "1 - Ввести значения вручную" << endl
+		 << "2 - Заполнить массив случайными числами" << endl;
+	cin >> iChoice;
+	while (iChoice < 1 || iChoice > 2) { 
+		cin >> iChoice;
+	}
+	
+	if (iChoice == 1) {
+		cout << "Введите значения в диапазоне [-10;10]:" << endl;
+		for (int i = 0; i < iSize; i++) {
+
+			int iValue = -11;
+			cin >> iValue;
+
+			while (iValue < -10 || iValue > 10) {
+				cout << "Введите значение в диапазоне [-10;10]" << endl;
+				cin >> iValue;
+			}
+
+			piArray[i] = iValue;
+		}
+	}
+
+	else if (iChoice == 2) {
+    srand(time(NULL));
+		for (int i = 0; i < iSize; i++) {
+			piArray[i] = rand() % 21 - 10;
+		}
+	}
+
+	vGetArray(iSize, piArray);
+}
+
+// Сумма элементов, имеющих нечетное значение
+// iSize - размер массива
+// piArray - указатель на массив
+void vSum(int iSize, int* piArray) {
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.